db/migrate/20151011151130_create_course_user_achievements.rb

```diff
@@ -0,0 +1,13 @@
+class CreateCourseUserAchievements < ActiveRecord::Migration
+  def change
+    create_table :course_user_achievements do |t|
+      t.references :course_user
+      t.references :achievement, foreign_key: { references: :course_achievements }
+      t.datetime :obtained_at, null: false
```

Comment:
There are 2 timestamps: one is when the student submits his/her mission and another is when the achievement is actually awarded i.e. the mission is graded by a tutor. Do I understand correctly @allenwq ?